sdk/identity/azure-identity/inc/azure/identity/managed_identity_credential.hpp

```diff
@@ -42,6 +47,34 @@ namespace Azure { namespace Identity {
     std::string ToString() const { return m_resourceId; }
```

Comment:
```suggestion
    std::string const& ToString() const { return m_resourceId; }
```

Reply:
I am curios, what's the benefit of this suggestion to return the string as `const&`?

We have other `ToString()` methods in the SDK, which all return `std::string` as is, so I was following an existing pattern.
https://github.com/Azure/azure-sdk-for-cpp/blob/4d829f32baff4757b0597b0e8b87ca5913d085f4/sdk/core/azure-core/inc/azure/core/uuid.hpp#L38
https://github.com/Azure/azure-sdk-for-cpp/blob/4d829f32baff4757b0597b0e8b87ca5913d085f4/sdk/core/azure-core/inc/azure/core/datetime.hpp#L191

That said, I do see the pattern you are suggesting elsewhere (http, etag, extensible enum).

Reply:
FWIW, from an architectural standpoint, returning a const& avoids a memory allocation and makes sense when the internal struct is a string type. 

I'm ambivalent on the use of this pattern, especially for items that are relatively small, but I strongly encourage its use for places where we return complex structures (like maps and vectors).

Reply:
Btw, technically this change is a breaking change, although it's unlikely to be significant (if someone attempted to directly modify the return value from the ToString call, their code would no longer compile because the return value is now const).

I would not block this change for that reason, but I would call it out in the changelog just in case.

Reply:
> FWIW, from an architectural standpoint, returning a const& avoids a memory allocation and makes sense when the internal struct is a string type.

Exactly. This does not apply to UUID and DateTime because their `ToString` methods create a brand-new string.

We could also add an rvalue reference-qualified overload of `ToString` but it might be overkill. 😅

```cpp
std::string&& ToString() && { return std::move(m_resourceId); }
```